vcGL/src/vcTextureHelper.cpp

```diff
@@ -319,22 +320,42 @@ udResult vcTexture_ConvertPixels(const void *pPixels, vcTextureFormat formatIn,
   return result;
 }
 
+struct vcTextureCallbackResult
+{
+  udFile *pFile;
+  const char *pFilename;
+  udResult result = udR_Success;
+};
+
+void vcTexture_StbCallback(void *context, void *data, int size)
+{
+  vcTextureCallbackResult *pData = (vcTextureCallbackResult *)context;
+  udResult writeResult = udFile_Write(pData->pFile, data, size);
```

Comment:
Write this directly into `pData->result`. Currently, if the last write fails you're returning success, which is wrong. Additionally, you don't initialise `pData->result` to anything, so if every write fails, it's undefined as to what happens.